src/NuGet.Core/NuGet.Build.Tasks.Pack/PackTask.cs

```diff
@@ -123,7 +124,36 @@ public override bool Execute()
                 if (string.IsNullOrEmpty(request.NuspecFile))
                 {
                     packageBuilder = logic.GetPackageBuilder(request);
+                    string noWarnProperties = null;
+
+                    if (packageBuilder?.Properties?.TryGetValue("NoWarn", out noWarnProperties) == true
+                        && !string.IsNullOrWhiteSpace(noWarnProperties))
+                    {
+                        PackageSpecificWarningProperties packageSpecificWarningProperties = new();
+
+                        foreach (string packageNoWarnProperty in noWarnProperties.Split(new string[] { Environment.NewLine },
+                            StringSplitOptions.RemoveEmptyEntries))
+                        {
+                            string[] packageNoWarnPropertyPars = packageNoWarnProperty.Split(';');
```

Comment:
`packageNoWarnPropertyPars` is misspelled, should be `packageNoWarnPropertyPairs`